src/packagers/JSConcatPackager.js

```diff
@@ -23,6 +24,25 @@ const helpers =
     .readFileSync(path.join(__dirname, '../builtins/helpers.js'), 'utf8')
     .trim() + '\n';
 
+function hasSideEffects(asset, {sideEffects} = asset.cacheData) {
```

Comment:
this function could probably go in `hoist.js`. That way `asset.cacheData.sideEffects` could always be a boolean computed using this function. Might be slightly faster, especially since we can cache the results.